steamclog/src/main/java/com/steamclock/steamclog/Destinations.kt

```diff
@@ -157,11 +160,51 @@ internal class ExternalLogFileDestination : Timber.DebugTree() {
     }
 
     private fun getExternalFile(): File? {
-        val date = SimpleDateFormat(fileNameTimestamp, Locale.US).format(Date())
-        val filename = "${fileNamePrefix}_${date}.${fileExt}"
+        val expiryMs = SteamcLog.config.autoRotateConfig.fileRotationSeconds * 1000 // defaults to 10 minutes
 
-        return try {
+        if (isCachedLogFileValid(expiryMs)) {
+            return cachedCurrentLogFile
+        }
+
+        val currentFile = findAvailableLogFile(expiryMs)
+        updateCachedLogFile(currentFile)
+        return currentFile
+    }
+
+    private fun isCachedLogFileValid(expiryMs: Long): Boolean {
+        // if you have a cached log file and you checked it within the expiry window
+        val now = Date().time
+        val cachedCurrentLogFile = cachedCurrentLogFile
+        val currentLogFileCachedTime = currentLogFileCachedTime
```

Comment:
why are you making a copy here?

Reply:
Because `currentLogFileCachedTime` and `cachedCurrentLogFile` are class level variables, the compiler can't smart-cast to a non-nullable version in the check below
```kotlin
return cachedCurrentLogFile != null &&
                currentLogFileCachedTime != null &&
                now - currentLogFileCachedTime < expiryMs
```

We assign a copy and check against that, since there's no way for the value to change in between the null check and the usage of the variable. We can safely remove the copy of `cachedCurrentLogFile` though, as the value isn't being used after the initial null check!

Reply:
@munl Removed the copy of `cachedCurrentLogFile`!